README.md

```diff
@@ -179,10 +179,167 @@ If you're calling a **GET** on `/entities` , you're asking for a list of entitie
+
+##### Examples
+
+###### Good
+
+> Getting comments from a post
+
+- **GET** `/posts/:id/comments`
+  The endpoint retrieves a list of comments for the post with the id you pass along in the url.
+
+> Getting the likes count of a post
+
+- **GET** `/posts/:id/like_count`
+  The endpoint retrieves the likes count of a post with the id you pass along in the url.
+
+> Deleting all the comments from a post
+
+- **DELETE** `/posts/:id/comments`
+  The endpoint deletes all the comments from the post with the id you pass along in the url.
+
+###### Bad
+
+> Getting a post
+
+- **GET** `/post/:id`
+  You're getting a post from a list/group of posts, so it should be plural `/posts/:id`.
+
+> Updating a post
+
+- **PUT** `/posts`
+  You're trying to update a post but you're not passing the id on the url. Passing it along in the request body is wrong.
+
+> Deleting a comment from a post
+
+- **DELETE** `/posts/:post_id/comment/:id`
+  Here you have a list/group of comments so it should be plural `/posts/:post_id/comments/:id`. Notice how I changed `comment` for `comments`.
+
+##### Reasoning
+
+Your endpoint name should be clear about the fact that it affects a single object or a group/list of objects. If you do a **POST** to `/posts` , you'll find the object you created by doing a **GET** to `/posts/:id`. 
+
+---
+
+#### Query String VS. Request Body. Where to send the object?
+
+> Query Strings are only valid on **GET** requests and every parameter they include is or should be used as a filter on the data returned. The only scenario where a parameter goes in the path is to identify a resource. Every other paremeter (in **POST**, **PUT**, **PATCH**, etc) should go in the request body. Any parameter that doesn't match any of the previous rules should go in headers (authentication, API versioning, etc).
+##### Examples
+
+###### Good
+
+> Getting posts for an specific user
+
+- **GET** `/posts?user_id=:id`
+  The endpoint retrieves a list of posts whose owner is the user with the id in the url
+
+> Rating a post
+
+- **PATCH** `/posts/:id`
+  `{"rate": 4.5}`
+  The endpoint gives the post a rating value. 
+
+> Geting places nearby the user
+
+- **GET** `/places?latitde=35.002&longitude=55.032`
+  The endpoint retrieves all the places nearby the user's location. Radius is a fixed value on the server.
+
+###### Bad
+
+> Creating a place
+
+- **POST** `/places?latitde=35.002&longitude=55.032&name=awesome%20place&type=restaurant`
+  Sending the parameters as a query string on a POST request is WRONG. They should be in the request body. Something like 
+```
+{
+	"latitude": 35.002
+	"longitude": 55.032
+	"name": "awesome place"
+	"type": "restaurant"
+}
+```
+
+> Updating a place
+
+- **PATCH** `/places?id=123&name=incredible%20place`
+  Don't send the parameters on the query string unless is a GET. You should be sending them in the request body like
+```
```

Comment:
add js to the right of the backticks to add color highlighting, like…
<pre>
```js
{
	"id": "123"
	"name": "incredible place"
}
```
</pre>

That looks like:
```js
{
	"id": "123"
	"name": "incredible place"
}
```